### ISSUE TITLE
"ignore-rhs" causes weird failures

### ISSUE THREAD

**Issue by pgsandstrom** [palantir/tslint#4746]
### Bug Report

- __TSLint version__: 5.16.0
- __TypeScript version__: 3.3.4000
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
let bool1:boolean = true
let bool2:boolean = true
let obj:{}|undefined = {}

const func = () => {
  if(bool1 && obj && bool2) {
    return 1
  } else {
    return 2
  }
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "strict-boolean-expressions": [true, "allow-undefined-union", "ignore-rhs"]
  }
}
```

#### Actual behavior

Get error `This type is not allowed in the operand for the '&&' operator because it unions more than one truthy/falsy type. Allowed types are boolean or undefined-union.`

#### Expected behavior

It should pass. If I remove "ignore-rhs" from tslint.json it passes.
